src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITValue.cpp

```diff
@@ -169,6 +202,40 @@ JITValuePointer LLVMJITValue::div(JITValue& rh) {
       getValueTypeTag(), parent_function_, ans, "div", false);
 }
 
+JITValuePointer LLVMJITValue::divWithErrorCheck(JITValue& rh) {
+  LLVMJITValue& llvm_rh = static_cast<LLVMJITValue&>(rh);
+  checkOprandsType(this->getValueTypeTag(), rh.getValueTypeTag(), "div");
+
+  llvm::Type* rh_llvm_type =
+      getLLVMType(llvm_rh.getValueTypeTag(), parent_function_.getLLVMContext());
+
+  auto zero_const =
+      rh_llvm_type->isIntegerTy()
+          ? makeJITValuePointer<LLVMJITValue>(
+                llvm_rh.getValueTypeTag(),
+                parent_function_,
+                llvm::ConstantInt::get(rh_llvm_type, 0, true),
+                "zero",
+                false)
+          : makeJITValuePointer<LLVMJITValue>(llvm_rh.getValueTypeTag(),
+                                              parent_function_,
+                                              llvm::ConstantFP::get(rh_llvm_type, 0.),
+                                              "zero",
+                                              false);
+
+  auto ifBuilder = parent_function_.createIfBuilder();
+  ifBuilder->condition([&zero_const, &llvm_rh]() { return llvm_rh.eq(zero_const); })
```

Comment:
seems` llvm_rh == 0` is  enough, no need for zero_const ? equal operator will do type infer.